addons/website/models/ir_qweb.py

```diff
@@ -104,6 +105,47 @@ def _post_processing_att(self, tagName, atts):
         if website and tagName == 'img' and 'loading' not in atts:
             atts['loading'] = 'lazy'  # default is auto
 
+        # If the cookie banner is activated, 3rd-party embedded iframes should
+        # be controlled. 'domains' is a watchlist on the iframe's src itself,
+        # 'classes' is a watchlist on container elements in which the iframes
+        # are/could be built on the fly client-side for some reason.
+        cookiesWatchList = {
+            # Checking 2nd-level domains mainly for Google (numerous subdomains
+            # & top-level domains)
+            'domains': [
+                "youtu", "youtube", "youtube-nocookie", # "youtu.be", "youtube.com", "youtube-nocookie.com",
+                "instagram", "ig", # "instagram.com", "ig.me",
+                "vimeo", # "player.vimeo.com", "vimeo.com",
+                "dailymotion", # "dailymotion.com",
+                "youku", # "player.youku.com", "youku.com",
+                "google", # "google.com", "google.be", "maps.google.com", ...
+                "facebook", "fb", # "facebook.com", "fb.com", "fb.me", "fb.watch",
+                "tiktok", # "tiktok.com",
+                "x", "twitter", "t", # "x.com", "twitter.com", "t.co",
+            ],
+            'classes': [
+                's_google_map',
+                's_map',
+                's_instagram_page',
+                'o_facebook_page',
+                'o_background_video',
+            ],
+        }
+        if website and website.cookies_bar \
+                   and not request.env['ir.http']._is_allowed_cookie('optional') \
+                   and not request.env.user.has_group('website.group_website_designer'):
+            iframeSrcInWatchlist = tagName == 'iframe' and 'src' in atts and atts['src'] \
+                    and any(domain in [d.lower() for d in url_parse(atts['src']).netloc.split('.')]
+                            for domain in cookiesWatchList['domains'])
```

Comment:
(see below comment about intersection)
```suggestion
           tagName == 'iframe' and atts['src']
                    and set(cookiesWatchList['domains']).intersection(url_parse(atts['src']).netloc.split('.'))
```